application/src/main/java/org/togetherjava/tjbot/features/help/HelpThreadAutoArchiver.java

```diff
@@ -133,20 +133,17 @@ private void handleArchiveFlow(ThreadChannel threadChannel, MessageEmbed embed)
         Supplier<RestAction<Message>> sendEmbedWithoutMention =
                 () -> threadChannel.sendMessageEmbeds(embed);
 
-        threadChannel.getGuild()
-            .retrieveMemberById(threadChannel.getOwnerIdLong())
-            .mapToResult()
-            .flatMap(foundMember -> {
-                if (foundMember.isSuccess()) {
-                    return sendEmbedWithMention.apply(foundMember);
-                }
-                logger.info(
-                        "Owner of thread with id: {} left the server, sending embed without mention",
-                        threadChannel.getId(), foundMember.getFailure());
-
-                return sendEmbedWithoutMention.get();
-            })
-            .flatMap(any -> threadChannel.getManager().setArchived(true))
-            .queue();
+        long authorId = helper.getAuthorByHelpThreadId(threadChannel.getIdLong()).orElseThrow();
+
+        threadChannel.getGuild().retrieveMemberById(authorId).mapToResult().flatMap(foundMember -> {
```

Comment:
odd naming, `foundMember` sounds like `boolean`, but its not. its also unclear which member ur looking for.
better would be `authorResults`

Reply:
This is from a PR from that was merged recently, but you're right it does sound like boolean. But the goal was to make it clear what it holds or not. Because it can or cannot have a member we're looking for.